### PR TITLE
refactor home page list, create new list form, add list creation succ…

### LINKED ISSUE #4: 4. As a user, I want to set up a new shopping list so I can start tracking purchased items.
## Summary 

A shopping list is a set of items associated with the owner's unique id and the name of the shopping list. Users need to be able to create new shopping lists. To do this, the `Home` view should present them with a form that allows them to enter the name of their list. Additionally, we need to do a few things to create that list:

1. Save the token to `localStorage`
2. Save the list to our database using the `createList` function in `firebase.js` 
3. Show the user the list view
## Acceptance criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x]  When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:
- [x] Clicking the button uses the `createList` function in `src/api/firebase.js` to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the `setListPath` function.
- [x] Once the list has been created and saved, the user is redirected to the `List` view.

## Notes
- The `createList` function takes **three** arguments: the current user's id, the current user's email, and the name of the shopping list.
- The `setListPath` function takes **one** argument: the new list token.
- If you get stuck trying to redirect the user, check out the [useNavigate](https://reactrouter.com/en/6.14.2/hooks/use-navigate) hook from `react-rotuer`.

### PR DESCRIPTION
**Pull request by aberger3647**
…ess message, redirect to /list

## Description

Based on a PR comment, we moved the singleList variable directly into the `<ul>` element. We created a form for creating a new list. After submitting the form, an error or success message populates below the form. If success, after 1 second, the user is redirected to the /list route.

## Related Issue

closes #4 

## Acceptance Criteria

UI-related tasks:

- [x] The Home view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic label element associated with it
- [x] The user can submit this form with both the mouse and the Enter key
- [x] When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:

- [x] Clicking the button uses the createList function in src/api/firebase.js to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the setListPath function.
- [x] Once the list has been created and saved, the user is redirected to the List view.

## Type of Changes

Feature add

### Before

Screenshot 2024-02-12 at 11 20 06 AM

### After

Screenshot 2024-02-12 at 1 19 34 PM

## Testing Steps / QA Criteria

Create list success:

- Type the name of a new list
- Click Submit or hit Enter
- There should be a message saying "List created"
- User should be redirected to /list

Create list error:

- Leave input blank
- Click Submit or hit Enter
- There should be a message saying "List not created"

### CODE REVIEW

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,28 +1,64 @@
...
+	const handleChange = (e) => {
+		setListName(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		try {
+			await createList(auth.currentUser.uid, auth.currentUser.email, listName);
+			setMessage('List created');
```
Comment (bdanzer) [inline src/views/Home.jsx]: I mentioned this in our call but I would add a comma like `List created, redirecting in 1 second...` so it's clear to the user.

File: src/views/Home.jsx [hunk 2]
```diff
@@ -1,28 +1,64 @@
...
+			setMessage('List not created');
+		}
+	};
+
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
+			<form onSubmit={(e) => handleSubmit(e)}>
```
Comment (jtkabenni) [inline src/views/Home.jsx]: As Britt mentioned we can probably just say `<form onSubmit={handleSubmit}>`

File: src/views/Home.jsx [hunk 3]
```diff
@@ -1,28 +1,64 @@
...
+				<label htmlFor="listName" name="listName">
+					List Name
+				</label>
+				<input
+					id="listName"
+					name="listName"
+					onChange={(e) => handleChange(e)}
```
Comment (jtkabenni) [inline src/views/Home.jsx]: As Britt mentioned we can probably just say `onChange={handleChange}`

**Review by flbarfield — APPROVED** [review]
Looks good! I like how you guys implemented the changes discussed on Tuesday! 

I now see how a message-less redirection would've been jarring as well.

File: src/views/Home.jsx [hunk 4]
```diff
@@ -1,28 +1,60 @@
...
-		);
-	});
+	const navigate = useNavigate();
+	const [listName, setListName] = useState('');
+	const [message, setMessage] = useState('');
+
+	const handleChange = (e) => {
+		setListName(e.target.value);
+	};
+	console.log(data);
```
Comment (bdanzer) [inline src/views/Home.jsx]: small nit: remove console.log

File: src/views/Home.jsx [hunk 5]
```diff
@@ -1,28 +1,60 @@
...
+		e.preventDefault();
+
+		try {
+			await createList(auth.currentUser.uid, auth.currentUser.email, listName);
+			setMessage('List created, redirecting in 1 second...');
+			setTimeout(() => {
+				navigate('/list');
+			}, '1000');
+		} catch (error) {
+			console.error(error);
```
Comment (bdanzer) [inline src/views/Home.jsx:27]: this console is fine since it's logging an error, but not always needed.

File: src/views/Home.jsx [hunk 6]
```diff
@@ -1,28 +1,60 @@
...
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="listName" name="listName">
+					List Name
+				</label>
+				<input id="listName" name="listName" onChange={handleChange}></input>
```
Comment (bdanzer) [inline src/views/Home.jsx]: nit: there is a quick way to close components that have nothing in-between them:

`<input id="listName" name="listName" onChange={handleChange}></input>`

to

`<input id="listName" name="listName" onChange={handleChange} />`

**Review by bdanzer — APPROVED** [review]
Looks good to merge! A few small nits mentioned.

**Comment by aberger3647** [comment]
Made those small changes mentioned by Britt!

**Pull request merged**